packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -353,6 +360,16 @@ export class LoggingError extends Error implements ILoggingError, Omit<IFluidErr
         }
     }
 
+    /**
+     * Determines if a given object is an instance of a LoggingError
+     * @param object - any object
+     * @returns - true if the object is an instance of a LoggingError, false if not.
+     */
+    public static typeCheck(object: any): object is LoggingError {
+        return hasTelemetryPropFunctions(object)
+            && typeof object?.errorInstanceId === "string";
```

Comment:
I've debated with myself about whether we want something more explicit than this.  Like checking for the private `_errorInstanceId`, or even renaming that field. Trouble is, because we support old loader versions for a long time we'd have to slowly stage that change - i.e. leave it like this for now but also rename `_errorInstanceId` to `_fluidErrorInstanceId`, and then in a year or something we can update this type guard.

I'm probably over-thinking it, what's your take? How likely is it and what would go wrong if someone else threw an error with `errorInstanceId` set to a string?

Reply:
I had the same concern with other people creating an error object that has the potential to get muddled by this typeCheck method. In terms of the ramifications, I think the main one I can see is:

If the Fluid framework and/or customers are handling external/internal errors in certain ways then the error handling logic could be broken if a customer provided error object is incorrectly type checked. In this case, 'certain ways' may or may not be critical functionality I suppose. 

Another thought is that if customers have automatic alarming setup off of errors then I would imagine an unknown/external error to be the type to set of their alarms. A "muddled' error object could bypass such a check because it would be incorrectly type checked as a LoggingError.

I'm going to take a look at symbols once again but I think even a simple unique key value would suffice. For example, if the all LoggingError's have a private instance variable 'type-uuid-xafsdfsdfsqwec' with value 'LoggingError-asdasdasd' I think we could both rest easy knowing external users would (almost) never have an object like that, unless purposefully created or extending LoggingError. - Do you have any thoughts on this approach?

Reply:
Let's continue this discussion in the task you opened in ADO, AB#1264.  For now, you can leave it as-is.